client/app/services/HelpTrigger.jsx

```diff
@@ -20,16 +31,121 @@ export default class HelpTrigger extends React.PureComponent {
     type: PropTypes.oneOf(Object.keys(TYPES)).isRequired,
   }
 
+  iframeRef = null
+
+  iframeLoadingTimeout = null;
+
+  constructor(props) {
+    super(props);
+    this.iframeRef = React.createRef();
+  }
+
+  state = {
+    visible: false,
+    loading: false,
+    error: false,
+  }
+
+  componentWillUnmount() {
+    this.onDrawerClosed();
+  }
+
+  get helpUrl() {
+    const [pagePath] = TYPES[this.props.type];
+    return DOMAIN + HELP_PATH + pagePath;
+  }
+
+  onIframeLoaded = () => {
+    const { contentWindow } = this.iframeRef.current;
+    contentWindow.postMessage(MESSAGES.HELP_DRAWER_REQUEST, DOMAIN);
+  }
+
+  onIframeError = () => {
+    this.setState({ error: true, loading: false });
+  }
+
+  onIframeReady = () => {
+    this.setState({ loading: false });
+    this.removeIframeListeners();
+  }
+
+  onPostMessageReceived = (event) => {
+    if (event.origin === DOMAIN && event.data === MESSAGES.HELP_DRAWER_READY) {
+      this.onIframeReady();
+    }
+  }
+
+  openDrawer = () => {
+    this.setState({ visible: true, loading: true, error: false });
+    setTimeout(this.onDrawerOpened, 300); // to prevent animation jank
```

Comment:
Yes, this is not ideal - but the simplest (and as risky as any other simple solution I could conceive). Open to suggestions.

Reply:
I'll create a PR for Ant with `onAfterOpen` perhaps.